enterprise/management/commands/backfill_role_assignments.py

```diff
@@ -0,0 +1,94 @@
+"""
+Management command for assigning enterprise_learner roles
+to existing linked enterprise users that are missing them.
+"""
+
+import logging
+from time import sleep
+
+from django.contrib import auth
+from django.core.management.base import BaseCommand
+
+from enterprise.constants import ENTERPRISE_LEARNER_ROLE
+from enterprise.models import EnterpriseCustomerUser, SystemWideEnterpriseRole, SystemWideEnterpriseUserRoleAssignment
+
+log = logging.getLogger(__name__)
+User = auth.get_user_model()
+
+
+class Command(BaseCommand):
+    """
+    Management command for assigning enterprise_learner roles to existing enterprise users.
+
+    Example usage:
+        $ ./manage.py backfill_role_assignments
+    """
+    help = 'Assigns enterprise_learner role to linked enterprise users missing them.'
+
+    def add_arguments(self, parser):
+        """
+        Entry point for subclassed commands to add custom arguments.
+        """
+        parser.add_argument(
+            '--batch-limit',
+            action='store',
+            dest='batch_limit',
+            default=250,
+            help='Number of records in each batch.',
+            type=int,
+        )
+
+        parser.add_argument(
+            '--batch-sleep',
+            action='store',
+            dest='batch_sleep',
+            default=5,
+            help='How long to sleep between batches.',
+            type=int,
+        )
+
+    def backfill_role_assignments(self, options):
+        """
+        Assigns enterprise_learner role to users.
+        """
+        batch_limit = options['batch_limit']
+        batch_sleep = options['batch_sleep']
+
+        role = SystemWideEnterpriseRole.objects.get(name=ENTERPRISE_LEARNER_ROLE)
+        ecus = EnterpriseCustomerUser.objects.select_related('enterprise_customer').filter(linked=True).iterator()
```

Comment:
[clarification] Should the batching apply to the querying of `EnterpriseCustomerUser` records? As is, we'll be querying for all of them at once, no? If so, is this OK?

Reply:
I'll reread the docs, but my thought that's where `.iterator()` is valuable, because you don't get the whole table at once

Reply:
it definitely doesn't _cache_ the whole table in the queryset, but now i'm doubting myself on what it does at the DB level

Reply:
`iterator()` will prevent crazy amount of memory being taken for the application, but not DB caching. That will still happen. I'll change this to be a more traditional batch